docs/docs/reference/value-providers.md

```diff
@@ -120,3 +120,38 @@ The following interpretation would create a node with the key `Hello, Joe!`:
         properties:
           age: !jmespath age
 ```
+
+## `!regex`
+
+The `!regex` value provider allows you to extract a value from a string using a regular expression. For example, if you wanted to extract
+the first name from a string given a record like this:
+
+```json
+{
+    "name": "Joe Smith",
+    "age": 25
+}
+```
+
+The following interpretation would create a node with the key `Joe`:
+
+```yaml
+- implementation: nodestream.interpreting:Interpreter
+  arguments:
+    interpretations:
+      - type: source_node
+        node_type: HelloNode
+        key:
+          first_name: !regex 
+            regex: "^(?P<first_name>[a-zA-Z]+)\s(?P<last_name>[a-zA-Z]+)$"
```

Comment:
I don't understand how `<last_name>` is used here... can you help me understand that? Seems like it would simply be dropped?

Reply:
Yeah you are right. This is just indicating that you can name groups and not use them all if that makes sense. Its only the group that you set in group that gets used.